### ISSUE TITLE
Add a unit test for salmontools extract-unmapped

### ISSUE THREAD

**Issue by jaclyn-taroni** [AlexsLemonade/refinebio#244]
### Context

Once we implement `salmontools extract-unmapped` (#83), we'll need a unit test.

### Problem or idea

The point of this tool is to essentially "filter" FASTQ files we give `salmon quant` to only those reads that Salmon could not map to the transcriptome index we gave it (e.g., unmapped reads). (We use the [`--writeUnmapped`](http://salmon.readthedocs.io/en/latest/salmon.html#writeunmappednames) option for `salmon quant` to get the read names.)

Some cursory thoughts:
* We could use the [sample data from SalmonTools](https://github.com/COMBINE-lab/SalmonTools/tree/474ba5947711074a015a268b47058ac8930cf566/sample_data).
* We could do an more extensive test (likely as part of a test of the entire RNA-seq pipeline) using simulated data. Specifically, if we use a tool like [Polyester](https://github.com/alyssafrazee/polyester) (as suggested by Rob Patro [here](https://github.com/AlexsLemonade/refinebio/issues/117#issue-296499941)) to simulate human reads and some _P. aeruginosa_ reads and combine them into a FASTQ file where we know the number of bacterial reads. We could check if the FASTA we get out of `salmontools extract-unmapped` is the set of bacterial reads we would expect. (This might be a future thought!)

### Solution or next step

To be revisited after #83 is accomplished.

### New Issue Checklist

- [x] The title is short and descriptive.
- [x] You have explained the context that led you to write this issue.
- [x] You have reported a problem or idea.
- [x] You have proposed a solution or next step.

**Comment by dongbohu** [comment]
Done by #250.
